### PR TITLE
Remove gridfs / blobstore package

### PR DESCRIPTION
**Pull request by SimonRichardson**
The blobstore package is no longer required, as we're using dqlite for metadata storage and file/s3 for blob storage.

The changeset removes the dependency on state altogether and just allows only file or s3 backends.

Unfortunately, we have a slight issue where we can't use dqlite in the state package (convention). This rears its head when using the object-store in tests. So for now, we're going to reply on an inmemory testing metadata service. This should be removed once dqlite is everywhere.

The default backend storage will be set to "file".

## Checklist

- [x] Code style: imports ordered, good names, simple structure, etc
- [x] Comments saying why design decisions were made
- [x] Go unit tests, with comments saying what you're testing

### File backed storage

```sh
$ juju boostrap lxd test --build-agent
$ juju add-model other
$ juju deploy ubuntu
```

```sh
$ juju bootstrap microk8s test
$ juju add-model other
$ juju deploy redis-k8s --channel=latest/edge
```

### S3 backed storage

Install docker or podman (I'll demonstrate using docker).

```sh
docker run -d \
   -p 9000:9000 \
   -p 9001:9001 \
   --user $(id -u):$(id -g) \
   --name minio1 \
   -e "MINIO_ROOT_USER=ROOTUSER" \
   -e "MINIO_ROOT_PASSWORD=CHANGEME123" \
   -v ${HOME}/.local/minio/data:/data \
   quay.io/minio/minio server /data --console-address ":9001"
```

Go to http://localhost:9001/access-keys and add an access key, keeping note of the key and secret.

Locate the IP for minio, we'll use that later.

```sh
$ juju bootstrap lxd test --build-agent --config="object-store-type=s3" --config="object-store-s3-endpoint=http://${IP_ADDRESS}:9000" --config="object-store-s3-static-key=${S3_KEY}" --config="object-store-s3-static-secret=${S3_SECRET}"
```

Once deployed you can inspect minio bucket.

```sh
$ juju enable-ha
$ juju add-model default
$ juju deploy juju-qa-test --resource foo-file="./tests/suites/resources/foo-file.txt"
```

## Links

**Jira card:** JUJU-5354

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -44,7 +44,6 @@ require (
 	github.com/hashicorp/vault/api v1.10.0
 	github.com/im7mortal/kmutex v1.0.1
 	github.com/juju/ansiterm v1.0.0
-	github.com/juju/blobstore/v3 v3.0.2
```
Comment (SimonRichardson) [inline go.mod:47]: So long, and thanks for all the fish.

**Review by manadart — COMMENTED** [review]
Nice chunk of deletions.

Couple of issues in _state/binarystorage/binarystorage_test.go_.

**Review by manadart — APPROVED** [review]
Sorry for the tardiness. You've got more conflicts to resolve, but this all looks good. 

QA 100%.

**Pull request merged**